cmd/agent/entrypoint.go

```diff
@@ -249,6 +257,68 @@ func (ep *Entrypoint) getReporterMetrics() map[string]interface{} {
 	return map[string]interface{}{"enabled-features": ep.cfg.EnabledFeatures}
 }
 
+func (ep *Entrypoint) supportHandler(rw http.ResponseWriter, r *http.Request) {
+	ep.mut.Lock()
+	cfg := ep.cfg
+	ep.mut.Unlock()
+
+	if cfg.DisableSupportBundle {
+		rw.WriteHeader(http.StatusNotFound)
+		_, _ = rw.Write([]byte("404 - support bundle generation is disabled"))
+		return
+	}
+	ep.sbmut.Lock()
+	defer ep.sbmut.Unlock()
+
+	duration := ep.srv.HTTPServer.WriteTimeout.Seconds()
+	if r.URL.Query().Has("duration") {
+		durationInt, err := strconv.Atoi(r.URL.Query().Get("duration"))
+		if err != nil {
+			http.Error(rw, fmt.Sprintf("duration value (in seconds) should be a positive integer: %s", err), http.StatusBadRequest)
+			return
+		}
+		if durationInt < 1 {
+			http.Error(rw, "duration value (in seconds) should be larger than 1", http.StatusBadRequest)
+			return
+		}
+		duration = float64(durationInt)
+		if duration > ep.srv.HTTPServer.WriteTimeout.Seconds() {
+			http.Error(rw, "duration value exceeds the server's write timeout", http.StatusBadRequest)
+			return
+		}
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(duration)*time.Second)
+	defer cancel()
+
+	ep.mut.Lock()
+	var (
+		enabledFeatures = ep.cfg.EnabledFeatures
+		httpSrvAddress  = ep.cfg.ServerFlags.HTTP.InMemoryAddr
+	)
+	ep.mut.Unlock()
+
+	// TODO(@tpaschalis) Can we also dynamically up the log_level while the
+	// support bundle is being generated?
+	var logsBuffer bytes.Buffer
+	logger := log.NewSyncLogger(log.NewLogfmtLogger(&logsBuffer))
+	defer func() {
+		ep.log.HookLogger.Enabled = false
+		ep.log.HookLogger.Logger = nil
+	}()
+	ep.log.HookLogger.Enabled = true
+	ep.log.HookLogger.Logger = logger
```

Comment:
This isn't goroutine safe; Enabled/Logger might be read by other goroutines (from different subsytems) at the same time this gets modified. 

You should add a method to the HookLogger to set these which needs a mutex (see next comment).